src/rgw/rgw_op.cc

```diff
@@ -4550,18 +4548,19 @@ void RGWBulkDelete::pre_exec()
 
 void RGWBulkDelete::execute()
 {
-  deleter = std::unique_ptr<Deleter>(new Deleter(store, s));
+  deleter = std::unique_ptr<Deleter>(new RGWBulkDelete::Deleter(store, s));
```

Comment:
@mattbenjamin: I think we can easily avoid the dynamic allocation here with `boost::optional`. It's needed due to the lifetime of `deleter` crossing borders of `execute` and `send_response` methods. From the other side, we cannot make `deleter` in constructor of `RGWBulkDelete` as the `req_state` isn't initialized then. Anyway, `boost::optional` should work for us.